
version: '3.8'

services:
  nginx:
    image: registry.glouton.int.nbyt.fr/nginx:latest
    volumes:
      - ./nginx.conf:/etc/nginx/nginx.conf
    ports:
      - "80:80"
    depends_on:
      public:
        condition: service_healthy
      api:
        condition: service_healthy
      db:
        condition: service_healthy


  public:
    build:
      context: .
      dockerfile: django-site/Dockerfile.front
    depends_on:
      db:
        condition: service_healthy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8000"]
      interval: 30s
      timeout: 10s
      retries: 5
    networks:
      - frontend

  api:
    build:
      context: .
      dockerfile: django-site/Dockerfile.api
    environment:
      - DJANGO_SUPERUSER_USERNAME=root
      - DJANGO_SUPERUSER_PASSWORD=password
      - DJANGO_SUPERUSER_EMAIL=myemail@example.com
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:8001"]
      interval: 30s
      timeout: 10s
      retries: 5
    depends_on:
      db:
        condition: service_healthy
    networks:
      - frontend

  db:
    image: registry.glouton.int.nbyt.fr/postgres:13
    environment:
      POSTGRES_DB: mydb
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: mypassword
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U myuser"]
      interval: 5s
      timeout: 10s
      retries: 20

networks:
  frontend:
    driver: bridge
  # backend:
  #   driver: bridge